### DISCUSSION TITLE
Mapping Coastal DDFs to Exisiting DDF/Occupancy Types

### DISCUSSION THREAD

**Discussion by rnugent3** [USACE/go-consequences#35] [Q&A]
go-consequences is attempting to develop a library that supports evaluation of damages to consequences receptors across multiple hazard types. For a structure, the process for computing consequences boils down to a single method "Compute(h hazards.HazardEvent)" where a HazardEvent can be a flood event, or a coastal event. This means that a structure is intended to be able to respond to either hazard. In order to do that we leverage the occupancy type to store damage functions for many hazards. 
Currently, the data source for structures is the National Structure Inventory. This means we are attempting to assign damage functions to the standardized occupancy types defined in the NSI schema. 
There are some rigidities to overcome in mapping coastal damage functions (PFRA and NACCS) to the existing 40 occupancy types. These rigidities come down to the existence of coastal damage functions unique to structure attributes (some existing in NSI, and some not) so that many coastal damage functions may potentially be mapped to a single existing occupancy type. 

1. Foundations
- PFRA no basement damage functions are broken out by deep and shallow foundation for waves, and are identical for saltwater inundation 
- Deep foundation refers to deep piles.
- Shallow foundation refers to short piers, crawlspaces, and slab-on-grade. 
- NSI Found_Type are: Slab, Pier, Unattached, and Basement. 
- Shallow foundation may refer to slab, pier, and unattached in the NSI, but no field entry currently exists for deep foundation. 

2. Basements
- PFRA breaks out basements by finished/unfinished 
- Unfinished basement basement damage includes cost of damaged water heaters, furnace, etc, but assumes no structural damage 
- Finished basement damage includes, in addition, the cost of interior partitions, insulation, wall, floor, and ceiling finishes, minor electrical, mechanical, and pluming work 
- This attribute does not exist in NSI

3. Commercial building type 
- NACCS commercial functions are broken out by engineered/non-engineered
- Engineered consists of steel frame with light cadding or precast infill and reinforced concrete
- Non-engineered consists of wood frame, steel or light metal, or steel with masonry infill
- NSI BlgdType consists of wood, manufactured (roughly non-engineered) and masonry, concrete bock, and steel (roughly engineered) 

4. Commercial contents
- NACCS commercial functions are broken out by perishable/nonperishable
- This does not exist in NSI, but could very roughly be applied to occupancy type based on SIC  (e.g. COM5, bank, non-perishable contents; COM6, hospital, perishable contents)

**Comment by HenryGeorgist** [comment]
Why can't we use Pier to map to deep foundation. What does Pier mean? Is that different than Pile?

**Reply by IdrisDobbs** [reply]
A pier is typically characterized by a discrete concrete footing or pad with columns and beams above. The beams carry the structural loads from the house to the posts/ piers which carry the load to the ground. The concrete pads are usually embedded to some depth below the frost line; usually 24 to 36" deep. 

Pile foundations are used in conditions where the soil bearing capacity is weak, susceptible to scour, and expansion or shrinkage. They are either driven to a hardened subsurface layer, or screwed deep when there is no hardened layer, and the friction of the surrounding material carries the building load.

https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.pinterest.com%2Fpin%2F435230751463295259%2F&psig=AOvVaw22qyT_0x2_BVBrjC4xxzbv&ust=1612970668069000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCKi9so-O3e4CFQAAAAAdAAAAABAD

**Reply by HenryGeorgist** [reply]
Great summary @IdrisDobbs - in your experience, is there a reliable source of information to drive the assumption of pile foundation types at a national scale as we develop plans for updates to NSI?

**Reply by IdrisDobbs** [reply]
That is difficult. I would say all else being equal, structure newness and closeness to erosion and wave action would be positively correlated with structures being on a pile supported foundation.

**Reply by HenryGeorgist** [reply]
Presence in a V zone? What proxy do we have for "closeness to erosion and wave action?"

**Reply by IdrisDobbs** [reply]
Perhaps. Usually the structures closest to the shoreline are the ones that are susceptible to wave attack and erosion damage. Using Beach-fx speak that would be the 1st row of houses

**Reply by IdrisDobbs** [reply]
I would also add that a foundation type is more relevant for determining erosion and wave attack than it is for inundation depth damage functions.

**Reply by HenryGeorgist** [reply]
I decided to run a quick test on the NSI, for the state of Florida (FIPS 12) 
```
func TestNsi_FL_FoundationTypes(t *testing.T) {
	f := []string{"12"}
	foundationTypes := make(map[string]int64)
	for _, ss := range f {
		GetByFipsStream(ss, func(str NsiFeature) {
			val, ok := foundationTypes[str.Properties.FoundType]
			if ok {
				v2 := val + 1
				foundationTypes[str.Properties.FoundType] = v2
			} else {
				foundationTypes[str.Properties.FoundType] = 1
			}

		})
	}
	fmt.Println(foundationTypes)
}
```
the NSI has this breakdown of foundation types for Florida:

- Basement: 1,315,130
- Crawl: 2,712,416
- Pier: 49,860
- Pile: 184,915
- Slab: 3,423,915
- Solid: 8,434

**Reply by IdrisDobbs** [reply]
Do you have a way of determining what the distribution is relative to the shoreline? Also, is there a distinction between a basement and a foundation type?

**Reply by GitLutz** [reply]
[FL.zip](https://github.com/USACE/go-consequences/files/5954301/FL.zip)
HAZUS identified every census block as either "Riverine" or "Coastal" (also, great lakes, but forget that). I don't have the details on the breakdown, but the areas I've looked at have been reasonable; it's not solely distance from shoreline, but seems to be somewhat more nuanced with coastal zones often extending up near rivers a few miles from the mouths.

In coastal areas, no structures are assumed to have basements. The rate of "Piles" increases in velocity zones.

The NSI randomly assigns from the attached distributions. But this will be improved in future versions.

**Reply by HenryGeorgist** [reply]
Which shoreline ;) ?  If a structure has a foundation type of basement we assume it has a basement, so not really, the basement would be the foundation type in that case. If it is not a basement, it could be any other type. This information is not our most accurate attribute in the NSI, but we can definitely improve it with time and better inputs.

**Reply by rnugent3** [reply]
It sounds like piles is an available foundation type, so that we can map deep to piles and shallow to crawl, pier, and and slab. I am not sure what a solid foundation is or where that would fall?

I wonder how strong of an assumption is no structures in coastal areas. I have come across a handful of coastal studies in the north east where a sizeable share of the inventory includes structures with basements.

**Reply by HenryGeorgist** [reply]
There are two things here I think:
1. proper assignment of damage functions -  for that, i like your proposal regarding deep and shallow, add solid foundation to shallow.
2. assumptions regarding the attribution of the NSI, that would not be something we would manage in the consequences engine, but rather in the next round of updating the NSI - so no action in this repository for that topic

**Comment by HenryGeorgist** [comment]
In terms of basements, we could combine the two with a uniform distribution for finished and unfinished and allow for random sampling.

**Reply by IdrisDobbs** [reply]
Basements are useful in instances where there is subterranean damageable value present. When you are talking about low lying areas with high water tables closer to the coast, you are unlikely to have subterranean value, and you are especially unlikely to have finished subterranean value.

**Reply by HenryGeorgist** [reply]
we know ground elevation - could we use that (some difference from sea level

**Reply by IdrisDobbs** [reply]
Something like that. The smaller the difference between the ground elevation and mean sea level, the less likely it is that the structure would have a basement?

**Reply by rnugent3** [reply]
I don't know about the rest of the country, but in line with what I wrote above, you are likely to see subterranean value in the north east in areas exposed to tidal inundation. Took a look at two recent studies, 1/3 structures has a basement in a coastal inventory in Jersey, and 1/4 in a coastal inventory in New York. 

I have not however seen any data on finished/unfinished.

**Comment by HenryGeorgist** [comment]
In terms of construction engineered vs non-engineered, it seems we do not have a clear identification of either - can we map an uncertainty distribution that averages them both using a uniform? I do not believe the construction type is universally dependable.

**Reply by IdrisDobbs** [reply]
I believe this would be the difference between structures with conditioned finished spaces vs structures with like sheds, garages, and pole barns that are more characterized by unfinished spaces. Perhaps occupancy types with the IND classification.

**Reply by IdrisDobbs** [reply]
I mean you could assign this damage function to NSI structures IND1 or IND2. Perhaps even COM10 if the asset isnt one of the large precast concrete multistory garage buildings.

**Reply by GitLutz** [reply]
In MMC studies, we typically associate "engineered" structures with steel buildings, like tall office buildings. But concrete garages would also make sense.

**Reply by IdrisDobbs** [reply]
I think it is a weakness in the terminology used in the NACCS.  I would argue that anything requiring a building permit for construction is likely an engineered structure. Whether it is a wood framed house, small industrial building, or a high rise. The difference is whether that structure has enclosed space that is air-conditioned, heated, with finished trim.

**Reply by rnugent3** [reply]
I am not sure that the label is as important as the assumptions made in developing the functions; one set, let's call that set A, represents steel frame with light cladding through reinforced concrete frame, another set, let's call that set B, consists of wood or light metal frame through steel frame with masonry infill.  I participated in an effort to map a set of detailed commercial and industrial occupancy types to A and B, and found that there was mix of A and B in a given occupancy type, and moved to an attempt to assign A and B at the structure level.

**Comment by HenryGeorgist** [comment]
I think we could take a stab at assigning the content curves by occupancy type - will you propose a map based on occupancy class?

**Reply by rnugent3** [reply]
I am happy to take a stab at a map.

**Comment by IdrisDobbs** [comment]
Hello all, I just joined the discussion thread.  We are just talking about coastal storm surge depth damage functions, correct?

**Reply by HenryGeorgist** [reply]
At the moment, though if you have experience to add on other hazards like wave action and scour we are open to the discussion. An option is to open a new thread on scour specifically.
The PFRA curves have the notion of wave height, with the classification of still water, medium intensitity 1-2.9 ft, and high intensity 3 ft+. That impacts which depth damage function gets assigned, but the driving parameter is still depth.

**Reply by IdrisDobbs** [reply]
I think it is a good idea to just focus on the storm surge inundation for now. For my information, what is PFRA?

**Reply by HenryGeorgist** [reply]
@IdrisDobbs Probable Flood Risk Assessment.

**Reply by IdrisDobbs** [reply]
I am unfamiliar with the PFRA. Where could I go to look that up?

**Comment by IdrisDobbs** [comment]
Some considerations:
Does the NSI values represent the total depreciated replacement value of the asset as a whole, or are the values reflective of damageable value? For example the NACCS urban high rise DDF assumes up to 25-26% of the structure value is at stake. It assumes this value is wrapped up in the 1st 10 floors I think. The important thing is there should be a relationship between how the structure is valued and the DDF.

It seems like for the NACCS functions, whether you use the minimum, most likely, or maximum DDF is dependent on foundation type, construction type, age, # stories and height above the 1st floor, etc.. You could create a "damage function matrix" that would assign a damage function by linking these factors from the NACCS function to the cooresponding factor in the NSI. This is similar to how damage functions are assigned in Beach-fx. For example,  Your matrix headings would be occupancytype, construction type, foundation type, age, number of stories, height above the 1st floor, 1st floor use. Pulling these criteria out of the NACCS Depth Damage Function Summary document tabular descriptions would be somewhat messy and there would be some transcription involved due to lack of uniformity in the tables, but it seems like it could be done.  (for example, Table 48 of the document in the following link)
https://www.nad.usace.army.mil/Portals/40/docs/NACCS/10A_PhysicalDepthDmgFxSummary_26Jan2015.pdf 
Would you be assigning a triangular distribution of damage functions to each record in the NSI or just one?

**Reply by HenryGeorgist** [reply]
The value is represented as depreciated replacement value @GitLutz can confirm. That seems like a bit of a limitation to the applicability of high rise damage functions that will need to be reconciled.

In https://github.com/USACE/go-consequences/blob/master/structures/occupancytypes.go we create a map of damage functions classified by hazard types and occupancy classifications, this is much like the matrix you are describing.

**Reply by GitLutz** [reply]
1) Structure values include (estimated) depreciated replacement values for the entire structures.
2) Content values are simple ratios of those structure values, no adjustments are made for what is or isn't damageable.
3) We'll definitely be re-thinking content ratios, etc. this year. But I think it'd be a mistake to include "damageable" in the value estimate. I think that would ideally be included in the damage functions themselves. 
   3b) For example, if a $100M office building is 100 stories tall then the damage function should recognize the likelihood of that value being uniformly distributed and that only a few stories may be at-risk of flooding, but more could be at-risk for other hazards.
   3c) Alternatively, we could add another col that makes estimates of the first 3 stories only.

**Reply by IdrisDobbs** [reply]
I concur. I think this is more of an issue with mid to high rise buildings with more than 10 stories. If you used the NACCS Urban High Rise damage function, it seems like you would need to adjust it somehow.

**Reply by HenryGeorgist** [reply]
The value of the structures should represent the depreciated replacement value. Other hazards like earthquake or fire will use the full valuation. Our damage curves should reflect the percentage of value lost relative to structure value, that way we can have a consistency in the valuation and vulnerability across hazards

**Reply by IdrisDobbs** [reply]
Right, So that would mean adjusting the NACCS damage function in those cases.

**Reply by HenryGeorgist** [reply]
@IdrisDobbs I took a moment to improve my comments on the test for the damage function family that allows us to be much more specific about hazard types and the associated damage function. Each occupancy type has a damage function family for structure and contents. That gives us tremendous flexibility so far (though it might need some more thought). We also can assign any damage function in the function family as a function with a distribution (to meet the EGM guidance for single family residential structures)
I added comments to the test to show how I am testing the capability and it passes my example tests. The application of the data structures is made more realistic in the occupancytypes.go file I mentioned before.

https://github.com/USACE/go-consequences/blob/587076c011864c08b8cef383200a2e14e537c084/structures/occupancytypes_test.go#L10

**Comment by IdrisDobbs** [comment]
This is the document that we tend to use for our non-residential depth damage functions. This is the IWR 2013 nonresidential DDFs based on expert ellicitation. 

[IWR_DepthDamageFunctions_NonResidential.docx](https://github.com/USACE/go-consequences/files/5954297/IWR_DepthDamageFunctions_NonResidential.docx)

**Reply by IdrisDobbs** [reply]
This is a spreadsheet that I put together awhile ago that links the type of structures to the DDF source. It isn't a neat map to the NSI occupancy type, but its a start. hope this stuff is helpful.

[DDF Sources.xlsx](https://github.com/USACE/go-consequences/files/5954390/DDF.Sources.xlsx)
